tracegen/internal/tracegen/config.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracegen
+
+import (
+	"flag"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/go-logr/logr"
+)
+
+// Config describes the test scenario.
+type Config struct {
+	Workers  int
+	Traces   int
+	Marshal  bool
+	Pause    time.Duration
+	Duration time.Duration
+	Service  string
+
+	// OTLP config
+	Endpoint string
+	Insecure bool
+}
+
+// Flags registers config flags.
+func (c *Config) Flags(fs *flag.FlagSet) {
+	fs.IntVar(&c.Workers, "workers", 1, "Number of workers (goroutines) to run")
+	fs.IntVar(&c.Traces, "traces", 1, "Number of traces to generate in each worker (ignored if duration is provided")
+	fs.BoolVar(&c.Marshal, "marshal", false, "Whether to marshal trace context via HTTP headers")
+	fs.DurationVar(&c.Pause, "pause", time.Microsecond, "How long to pause before finishing trace")
```

Comment:
> It would allow to generate specified number of traces per unit of time

Rate then ?

```go
fs.IntVar(&c.Rate, "rate", 1, "the number of traces per second to generate")
```

Reply:
Rate works too.